src/crosschainMap/crosschainMap.ts

```diff
@@ -47,6 +50,19 @@ export class CrosschainMap {
   public getTokenIDByAddress = ({ address, chainId }: TokenBaseParams) =>
     this.tokenMapByChain[chainId][address.toLowerCase()]?.id
 
+  public getCrosschainTokenBySymbol = ({
```

Comment:
Isn't it dangerous to get token by symbol as symbols aren't unique?

Reply:
I have written some logic to get tokens by symbol and add address. FYI

https://github.com/levelkdev/swapr-dapp/blob/4396933512f95dc55b23cdc4bf72900366021715/src/state/lists/hooks.ts#L51-L95

Reply:
I don't know in depth but at first sight I would say "yes, this can def. be abused".
I need to get same tokens in different chains.
The common info is

- name: string
- symbol: string
- decimals: number
- logoURI?: string
- chainId: SupportedChains
- address: string

I am not sure we can use as the I would expect the name to have the same problem as the symbol and the other fields to be either useless or chain specific giving me nothing to work with.
The purpose of this function is to mitigate a gap of tokens from the Optimism pool and to match the information we get with what we already have from the other pools. I am not sure how to address this problem.
I am open to suggestions from anyone on this.